src/main/scanImagesProcess.test.ts

```diff
@@ -0,0 +1,96 @@
+import 'jest-extended';
+import fs from 'fs-extra';
+import { createMockServiceContainer } from '../test/mockCreators';
+import { createStore } from './createStore';
+import md5 from 'md5';
+import { scanImages, getImageStats } from './scanImagesProcess';
+
+const sitePath = '/Users/cool-man-joe/Local Sites/twice-baked-potato';
+const serviceContainer = createMockServiceContainer(sitePath);
+
+const imageDataStore = createStore();
+
+jest.mock('fs-extra');
+fs.statSync.mockImplementation((filePath: string) => ({
+	size: Math.floor(Math.random() * 99999) + 700
+}));
+
+
+const mockFilePaths = [
+	'fake/path/file1.jpeg',
+	'fake/path/file2.jpeg',
+];
+
+const mockImageData = {
+	'40cf40c608bab653903bec92ab8bd26a': {
+		originalImageHash: '40cf40c608bab653903bec92ab8bd26a',
+		compressedImageHash: 'snthsnthsnthstnh',
+		filePath: 'fake/path/file1.jpeg',
+		originalSize: 10000000,
+		compressedSize: 5000000,
+		fileStatus: 'succeeded',
+		isChecked: true,
+	},
+	'b47c61eec6e37403c6be0cea67aa7a8e': {
+		originalImageHash: 'b47c61eec6e37403c6be0cea67aa7a8e',
+		compressedImageHash: 'hdhdhdhdhdhdhdhdhd',
+		filePath: 'fake/path/file2.jpeg',
+		originalSize: 7000000,
+		compressedSize: 3500000,
+		fileStatus: 'succeeded',
+		isChecked: true,
+	}
+}
+
+// mock out utils so that we don't accidentally make calls to fs, etc.
+// also this makes assertions much easier and utils should be tested independantly anyways
+jest.mock('./utils');
+const utils = require('./utils');
+utils.getImageFilePaths.mockImplementation(() => {
+	return mockFilePaths;
+});
+utils.getFileHash.mockImplementation((filePath) => {
+	return md5(filePath);
+});
+utils.getImageIfCompressed.mockImplementation((fileHash, imageData) => {
+	if(fileHash === '40cf40c608bab653903bec92ab8bd26a') {
```

Comment:
It's definitely not a deal breaker, but I like to store complex strings (or even object keys) that are used multiple times like this in variables to help eliminate syntax mistakes down the road.

Reply:
✅